[Feat] 홈화면 Compose 마이그레이션

etama123 opened the pull request:
## #️⃣ 이슈 번호

> #11 

<br>

## 🛠️ 작업 내용

- 홈화면을  Compose로 마이그레이션했습니다. 


> [마이그레이션 순서]
> 1. 각 아이템을 별도의 컴포넌트로 구성
> 2. HomeScreen 내부에 FestivalOverview 내부에 LazyColumn을 이용해 리스트 구성
> 

<br>

## 🙇🏻 중점 리뷰 요청

### 주의할 부분 
- 설정 화면 > homeViewModel 코드를 수정하면서, 현재 접속 중인 학교를 주석 처리해두었습니다🙏
- 실행 시에, 포스터 이미지가 placeholder 이미지로 보이고 있습니다. 이는 이미지 표시할 때, CoilImage 컴포저블을 사용했고, 아직 해당 컴포저블에 dev와 prod 이미지 주소를 다르게 읽는 부분이 반영되지 않아 그렇습니다. 

### 논의할 부분 
**1. 홈 화면 로딩 관련**
모든 상태(festival, lineup 등등)이 성공 상태일 때, 홈 화면 전체를 보여주는 게 좋을 까요? 아니면 각 항목별로 로딩 상태일 때의 액션(스켈레톤 등)을 추가하는 게 좋을까요?

**2. FestabookColor.white**
해당 속성이 기존에 사용하던 배경색(진짜 쌩 화이트)가 아니라 새롭게 지정한 디자인 시스템의 white 입니다. 따라서 화면의 배경색을 지정할 때,
FestabookColor.white를 사용하지 못하고 Color.white를 사용해야합니다. 이 점에 대해 따로 배경색을 추가하는 방향은 어떤 지 논의하고 싶습니다. 

<br>

## 📸 이미지 첨부 (Optional)

<img src="파일주소" width="50%" height="50%"/>


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

## 릴리스 노트

* **New Features**
  * Compose 기반 UI 컴포넌트 추가: 홈 화면 헤더, 포스터 리스트, 라인업 항목, 축제 정보 등 새로운 화면 구성 요소 도입

* **Refactor**
  * 전통적 XML 기반 레이아웃에서 Jetpack Compose 기반 UI로 마이그레이션
  * 상태 관리를 LiveData 기반에서 Flow 기반으로 변경하여 비동기 처리 개선

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

etama123 committed:
feat(home): HomeHeader 컴포저블 추가

etama123 committed:
feat(Home): HomeFestivalInfo 컴포저블 추가

etama123 committed:
feat(home): HomeLineupHeader 컴포저블 추가

etama123 committed:
feat(home): HomePosterList 컴포저블 구현 및 무한 스크롤 페이저 적용

etama123 committed:
feat(home): 홈 화면 아티스트 및 라인업 아이템 컴포저블 구현

etama123 committed:
feat(home): 홈 화면 Compose 전환 및 관련 로직 리팩토링

parkjiminnnn commented on app/src/main/java/com/daedan/festabook/presentation/home/HomeFragment.kt:
```diff
@@ -1,184 +1,49 @@
-import timber.log.Timber
 
-@ContributesIntoMap(scope = AppScope::class, binding = binding<Fragment>())
+@ContributesIntoMap(scope = AppScope::class, binding = binding<androidx.fragment.app.Fragment>())
```
임포트를 안하고 사용하신 이유가 있으실까요?

etama123 replied:
😅 수정했습니다

parkjiminnnn commented on app/src/main/java/com/daedan/festabook/presentation/home/HomeViewModel.kt:
```diff
@@ -1,32 +1,35 @@
 @ContributesIntoMap(AppScope::class)
 @ViewModelKey(HomeViewModel::class)
 class HomeViewModel @Inject constructor(
     private val festivalRepository: FestivalRepository,
```
클래스에 `@Inject`를 붙이는걸로 변경하는게 좋을 것 같아용

parkjiminnnn replied:
이 부분 아직 반영이 안되어있네요~

parkjiminnnn commented on app/src/main/java/com/daedan/festabook/presentation/main/MainActivity.kt:
```diff
@@ -158,8 +163,13 @@ class MainActivity :
+                homeViewModel.navigateToScheduleEvent.collectLatest {
+                    binding.bnvMenu.selectedItemId = R.id.item_menu_schedule
+                }
+            }
```
`collectLatest `를 사용하신 이유가 궁금합니다.

etama123 replied:
화면 이동과 같은 버튼에서 `collectLatest`가 마지막 요청만 처리하게 되어 중복 동작을 예방할 수 있다고 알고 있어 `collectLatest`을 사용했어요!

parkjiminnnn replied:
음 선택된 아이템을 바꾸는 동기로직은 바로 호출되는 순간 바로 수행되어서, 요청을 받는 중간(suspend 될 때)에 새로운 값이 수집되면 이전 작업을 버리는 `collectLatest`는 의미가 없다고 생각하는데 타마의 생각은 어떠신가요?

etama123 replied:
해당 코드 내부에 중단점이 없어, `collectLatest`가 의미없는 사용이었겠네요😥
다른 방어 로직으로 변경해봤습니다!!

parkjiminnnn commented on app/src/main/java/com/daedan/festabook/presentation/home/component/HomeFestivalInfo.kt:
```diff
@@ -0,0 +1,52 @@
+) {
+    Column(
+        modifier = modifier.fillMaxWidth(),
+    ) {
+        Text(
+            text = festivalName,
+            style = FestabookTypography.displayMedium,
+            color = FestabookColor.black,
+            modifier = Modifier.padding(horizontal = 20.dp),
+        )
+
+        Spacer(modifier = Modifier.height(8.dp))
+
+        Text(
+            text = festivalDate,
+            style = FestabookTypography.bodyLarge,
+            color = FestabookColor.gray500,
+            modifier = Modifier.padding(horizontal = 20.dp),
+        )
+    }
+}
```
`Text`에서 `padding`을 주지 않고 `Column`에서 한 번에 주는건 어떄요?

parkjiminnnn replied:
이 부분도 아직 반영이 안되어 있어요

parkjiminnnn commented on app/src/main/java/com/daedan/festabook/presentation/home/component/HomeHeader.kt:
```diff
@@ -0,0 +1,68 @@
+            Image(
+                painter = painterResource(id = R.drawable.ic_dropdown),
+                contentDescription = stringResource(id = R.string.home_navigate_to_explore_desc),
+            )
```
Icon함수가 아닌 Image함수를 사용하신 이유가 궁금해요

etama123 replied:
해당 부분도 다른 부분 처럼 Icon 함수를 사용하도록 수정했습니다

parkjiminnnn commented on app/src/main/java/com/daedan/festabook/presentation/home/component/HomeLineupHeader.kt:
```diff
@@ -0,0 +1,78 @@
+        modifier =
+            modifier
+                .fillMaxWidth()
+                .padding(start = 16.dp, top = 16.dp, end = 16.dp, bottom = 16.dp),
```
16.dp 만 명시해줘도 좋을 것 같아용

parkjiminnnn commented on app/src/main/java/com/daedan/festabook/presentation/home/component/HomeLineupHeader.kt:
```diff
@@ -0,0 +1,78 @@
+                    .clickable(
+                        onClick = onScheduleClick,
+                    )
+                    .padding(4.dp),
```
요 친구 있으나 없으나 별 차이는 없어보이긴하는데 어떻게 생각하시나용?

parkjiminnnn commented on app/src/main/java/com/daedan/festabook/presentation/home/component/HomeLineupItem.kt:
```diff
@@ -0,0 +1,150 @@
+    ) {
+        // 날짜 + 배지 영역
+        Row(
+            modifier = Modifier.padding(horizontal = 16.dp),
```
어차피 Spacing이 정의되어 있어서 중요한건 아니긴 한데 이거랑 아래의 `LazyRow`의 `contentPadding`에 16.dp를 안주고 외부에서 한 번에 주는게 제어하기 더 쉽지 않을까 하는데 어떻게 생각하시나요?

etama123 replied:
외부에서 padding을 주게 되면,
가로 스크롤 시에 왼쪽에 여백이 생긴채로 스크롤 되게 되더라구요.
그래서 `contentPadding`으로 초기 여백값을 설정했어요!

혹시 다른 방법이 있을까요?

parkjiminnnn replied:
아하 무슨 말씀인지 이해했습니다~ 저도 이대로 가는 방법밖에는 안떠오르네요

parkjiminnnn commented on app/src/main/java/com/daedan/festabook/presentation/home/component/HomeScreen.kt:
```diff
@@ -0,0 +1,196 @@
+    Scaffold(
+        modifier = modifier.fillMaxSize(),
+        containerColor = Color.White,
+    ) {
```
슬렉에서 말씀하신 빨간줄이 여기군용.
`Scaffold`를 사용하고 있는데 `content`에서 `PaddingValues`를 사용하지 않고 있어서 빨간줄 뜨는걸로 알고 있습니다.

etama123 replied:
오잉 
제 화면에서 이 부분은 에러가 나지 않는데, 혹시 에러가 나나요?!

parkjiminnnn replied:
에러는 아니고 경고창이 보입니당. innerPadding으로 LazyColumn의 padding에 값을 주는 것도 좋을 것 같네요

parkjiminnnn commented on app/src/main/java/com/daedan/festabook/presentation/home/component/HomePosterList.kt:
```diff
@@ -0,0 +1,112 @@
+                        scaleY = scale
+                        this.alpha = alpha
+                    }
+                    .clip(RoundedCornerShape(10.dp)),
```
공통컴포저블인 `cardBackground`를 사용하는 것도 좋아보이네용

parkjiminnnn commented on app/src/main/java/com/daedan/festabook/presentation/home/component/HomeScreen.kt:
```diff
@@ -0,0 +1,196 @@
+            // 하단 여백 추가
+            item {
+                Box(modifier = Modifier.padding(bottom = 60.dp))
+            }
```
다른건 전부 `Spacer`였는데 여기는`Box`로 하신 이유가 있으실까요?

etama123 replied:
해당 영역에 나중에 앱 정보와 같은 내용이 들어갈 수 있겠다고 생각했는데, 아직 고려하지 않아도 될 것 같네요😅

parkjiminnnn commented on app/src/main/java/com/daedan/festabook/presentation/home/component/HomeScreen.kt:
```diff
@@ -0,0 +1,196 @@
+            // 라인업 리스트
+            when (lineupUiState) {
+                is LineupUiState.Success -> {
+                    val lineups = lineupUiState.lineups.getLineupItems()
```
여기 프로퍼티가 아니라 `getLineupItems`함수로 해야했던 이유가 있었을까요?(진짜 모름)

etama123 replied:
`LineupItemUiModel data clas`s를 확인하면, 오늘이 라인업의 날짜와 같은 날인 지 판단하는 로직이 있습니다!
해당 로직은 이 함수를 호출할 때(UI가 이 데이터를 그리는 시점에) 계산이 되어야하는데요. 그래서 `getLineupItems` 함수를 사용했습니다.

etama123 replied:
그런데 구조를 다시 생각해봤을 때, ui에서 계산 로직을 불러는 과정이 어색하게 느껴져 viewModel 내부로 계산 로직을 옮겼습니다~

parkjiminnnn commented on app/src/main/java/com/daedan/festabook/presentation/home/component/HomeLineupItem.kt:
```diff
@@ -0,0 +1,150 @@
+                    .width(75.dp)
+                    .height(1.dp)
+                    .background(FestabookColor.gray700),
+        )
```
`HorizontalDivider`사용하는건 어떠신가요?

프리뷰를 봤을 때 오늘이면 divider와 날짜+오늘 텍스트의 width가 미세하게 다릅니다!
divider와 텍스트를 하나의 컴포저블로 묶어서 사이즈 관리를 하면 수월하지 않을까 합니다!

etama123 replied:
헉,, 예리하시네요😯
바로 수정했습니다. `.width(IntrinsicSize.Max)` 이 속성도 알게되었네요!

parkjiminnnn reviewed (changes requested):
고생하셨습니다 타마
궁금한 점 코멘트 남겨봤습니다 확인 부탁드려요.
그리고 PR 제목 수정 부탁드려요..ㅋㅋ

> 1. 홈 화면 로딩 관련
> 모든 상태(festival, lineup 등등)이 성공 상태일 때, 홈 화면 전체를 보여주는 게 좋을 까요? 아니면 각 항목별로 로딩 상태일 때의 액션(스켈레톤 등)을 추가하는 게 좋을까요?
> 
> 2. FestabookColor.white
> 해당 속성이 기존에 사용하던 배경색(진짜 쌩 화이트)가 아니라 새롭게 지정한 디자인 시스템의 white 입니다. 따라서 화면의 배경색을 지정할 때,
> FestabookColor.white를 사용하지 못하고 Color.white를 사용해야합니다. 이 점에 대해 따로 배경색을 추가하는 방향은 어떤 지 논의하고 싶습니다.

1. 분리하는게 좋을 것 같아요
2.  Color.white를 사용해야만 하는 이유가 뭔가요?
Scaffold는 색들이 자동으로 materialTheme을 따라가서 저희 FestabookTheme 색 정의에 대한 논의를 할 때 얘기해보면 좋을 것 같아용.

etama123 commented:
@coderabbitai review

etama123 commented:
> 2\. Color.white를 사용해야만 하는 이유가 뭔가요?
>     Scaffold는 색들이 자동으로 materialTheme을 따라가서 저희 FestabookTheme 색 정의에 대한 논의를 할 때 얘기해보면 좋을 것 같아용.

네! 말씀하신 것 처럼 Scaffold에 containerColor을 지정하지 않으면 MaterialTheme의 기본 색상을 따라가더라구요! 
그래서 Color.white로 색상을 지정해두었습니다. 
 FestabookTheme에 지정하면 해결될 것 같군요😁

etama123 committed:
refactor(home): 홈 화면 컴포저블 및 코드 정리

oungsi2000 commented on app/src/main/java/com/daedan/festabook/presentation/home/component/HomeArtistItem.kt:
```diff
@@ -0,0 +1,70 @@
+object HomeArtistItem {
+    val ArtistImage = RoundedCornerShape(
+        topStartPercent = 50,
+        topEndPercent = 50,
```
private 을 다는건 어떻게 생각하시나요?

oungsi2000 commented on app/src/main/java/com/daedan/festabook/presentation/home/component/HomeArtistItem.kt:
```diff
@@ -0,0 +1,70 @@
+
+@Preview
+@Composable
+private fun HomeArtistItemPreview() {
+    HomeArtistItem(
+        artistName = "실리카겔",
```
이건 TMI일 수도 있는데, Theme으로 감싸주지 않고 Preview를 실행하는건 실제 사용할 때와 다르게 동작할 수 있어 주의해야 하는 것으로 알고있어요 !

oungsi2000 commented on app/src/main/java/com/daedan/festabook/presentation/home/component/HomeHeader.kt:
```diff
@@ -0,0 +1,68 @@
+                text = schoolName,
+                style = FestabookTypography.displayLarge.copy(
+                    platformStyle = PlatformTextStyle(includeFontPadding = false),
+                    lineHeight = 34.sp
```
참고해주세요 !

oungsi2000 commented on app/src/main/java/com/daedan/festabook/presentation/home/component/HomeHeader.kt:
```diff
@@ -0,0 +1,68 @@
+                color = FestabookColor.black,
+            )
+
+            Spacer(modifier = Modifier.width(4.dp))
+
```
festabookSpacing을 써도 될 것 같아요 !

oungsi2000 commented on app/src/main/java/com/daedan/festabook/presentation/home/component/HomeLineupHeader.kt:
```diff
@@ -0,0 +1,78 @@
+                painter = painterResource(id = R.drawable.ic_arrow_forward_right),
+                contentDescription = null,
+                tint = FestabookColor.gray400,
+                modifier = Modifier.size(12.dp),
```
기존 XML에서도 contentDescription이 없었을까요?
Compose는 XML과 달리 UI의 Sementic을 중요시하는 것으로 알고 있어서 없다면 추가해줘도 좋을 것 같아요!

oungsi2000 commented on app/src/main/java/com/daedan/festabook/presentation/home/component/HomeLineupItem.kt:
```diff
@@ -0,0 +1,150 @@
+
+        Spacer(modifier = Modifier.height(4.dp))
+
+        Box(
```
👀

oungsi2000 commented on app/src/main/java/com/daedan/festabook/presentation/home/component/HomeLineupItem.kt:
```diff
@@ -0,0 +1,150 @@
+
+        // 아티스트 가로 리스트
+        LazyRow(
+            contentPadding = PaddingValues(horizontal = 16.dp),
```
요기도 festabookSpacing 써줘도 좋을 것 같아요 !

oungsi2000 commented on app/src/main/java/com/daedan/festabook/presentation/home/component/HomePosterList.kt:
```diff
@@ -0,0 +1,112 @@
+                contentDescription = null,
+                modifier = Modifier.fillMaxSize(),
+            )
+        }
```
### 중요

기존에는 포스터 클릭 시 확대 가능한 Dialog를 띄웠는데요 (엄재웅님 라이브러리) 해당 기능이 누락된 것 같습니다 !

etama123 replied:
기존에 사용하던 GetStream/photoview-android 라이브러리는 XML 기반 라이브러리이기 때문에, Compose환경에서는 AndroidView로 감싸서 사용해야하더라구요.
그래서 다른 라이브러리 사용을 슬랙-#기능제안에서 제안하고자합니다!

oungsi2000 commented on app/src/main/java/com/daedan/festabook/presentation/home/component/HomePosterList.kt:
### 중요

포스터 이미지가 없는 축제에 들어갔을 때, (dev서버의 가천대학교) 
기존에는 자동으로 에러 이미지가 채워졌는데, 
현재는 포스터 영역이 완전히 사라집니다 !

etama123 replied:
저도 이 부분에 대해서 고민해봤는데요. 
포스터 이미지가 없는데, 포스터 영역이 화면에 있는 게 맞나? 이런 생각도 들더라구요. 
어떻게 생각하시나요!?
@parkjiminnnn 의 의견도 궁금합니다!

parkjiminnnn replied:
뭔가 없는 이미지라도 ui는 그대로 채우는게 자연스러울거라고 생각하는데 에러이미지로 없는 이미지를 채우자니 뭔가 좀 딱딱한 느낌이라..
Festabook 로고가 들어간 이미지라도 간단하게 만들어서 채우는게 보기 좋지 않을까 합니다.

oungsi2000 reviewed (changes requested):
마이그레이션 고생하셨습니다 !
포스터 스크롤이 더 부드러워진 느낌도 드네요. 
전반적으로 넘 잘해주셨지만 몇 가지 버그가 있어 RC 드립니다 !

>모든 상태(festival, lineup 등등)이 성공 상태일 때, 홈 화면 전체를 보여주는 게 좋을 까요? 아니면 각 항목별로 로딩 상태일 때의 액션(스켈레톤 등)을 추가하는 게 좋을까요?

저도 분리하는 것이 좋을 것 같습니다 !
현재는 포스터에 스켈레톤이 없어 포스터가 로딩되기 전에 라인업이 로딩이 완료되면
포스터의 영역이 사라졌다 생기는 현상이 있어 조금 부자연스러운 것 같아요 !
스켈레톤으로 메꿔주면 훨씬 좋을 것 같습니다.

etama123 committed:
refactor(Home): 홈 화면 UI 컴포즈 컴포넌트 리팩토링 및 스타일 개선

홈 화면을 구성하는 주요 컴포저블(`HomeHeader`, `HomePosterList` 등)의 파라미터 네이밍을 정리하고, 디자인 시스템 및 커스텀 Modifier를 적용하여 코드 일관성을 높였습니다.

- **`HomeHeader.kt` 수정:**
    - 파라미터 이름을 `schoolName`에서 `universityName`으로 변경하여 도메인 모델과의 일관성을 맞췄습니다.
    - 드롭다운 아이콘을 `Image`에서 `Icon`으로 변경하고, `FestabookColor`와 고정 크기(24.dp)를 적용했습니다.

- **`HomePosterList.kt` 수정:**
    - 포스터 카드에 커스텀 Modifier인 `cardBackground`를 적용하여 배경 스타일을 개선했습니다.

- **`HomeScreen.kt` 수정:**
    - `HomeHeader` 호출 시 변경된 파라미터 명(`universityName`)을 반영했습니다.
    - 리스트 하단의 여백 처리를 위해 의미상 더 적절한 `Spacer`를 사용하도록 변경했습니다.

- **`HomeLineupHeader.kt` 수정:**
    - 일정 클릭 영역(`Row`)에서 불필요한 `padding(4.dp)`을 제거하여 레이아웃을 정돈했습니다.

etama123 committed:
refactor(Home): LineupUiState 데이터 구조 변경

etama123 committed:
refactor(HomeLineupItem): 날짜 영역 레이아웃 및 디바이더 구현 방식 변경

etama123 committed:
refactor(Home): 홈 화면 문자열 리소스 정리 및 접근성 개선

parkjiminnnn reviewed (approved):
고생 많으셨습니다. 몇 가지 코멘트만 확인 부탁드려요~

oungsi2000 reviewed (approved):
라이브러리 사용 제안이 엄재웅님 라이브러리 때문이었군요..!
근데 제안주신 라이브러리에 피치 줌 기능이 들어가 있는지는 제가 확인하지 못했네요.

이것만 반영해주시면 될 것 같아요!
고생하셨습니다 타마~~

etama123 committed:
refactor(HomeViewModel): `@Inject` 어노테이션 위치 수정

etama123 committed:
refactor(MainActivity): BottomNavigationView의 중복 선택 방지 및 flow 수집 방식 변경

etama123 committed:
refactor(HomeFestivalInfo): Column 내부 padding 적용 및 개별 Text padding 제거

etama123 committed:
build: Landscapist 라이브러리 의존성 추가

etama123 committed:
feat(common): FestabookImage 공통 컴포저블 구현

etama123 committed:
feat(home): 홈 화면 UI 상태 관리 로직 개선 및 `FestabookImage` 적용

oungsi2000 commented on app/src/main/java/com/daedan/festabook/presentation/common/component/FestabookImage.kt:
```diff
@@ -0,0 +1,160 @@
+) {
+    Dialog(
+        onDismissRequest = onDismiss,
+        properties = DialogProperties(usePlatformDefaultWidth = false)
```
public 으로 열어두는게 더 확장성이 높을 것 같은데 어떻게 생각하시나요?

etama123 replied:
해당 컴포저블을 열어 두게 되면 외부에서 이미지 컴포저블을 사용할 때 오히려 혼동이 생길 수도 있겠다고 생각해요!! 
그래서 이미지 컴포저블을 사용하는 통로를 하나로 제한하고 대신 기존 festabookimage에서 이게 팝업화면으로 띄울 건지 아닌지 속성으로 지정하도록 한 거였습니다!😳

oungsi2000 reviewed (approved):
긴 시간 동안 고생 많으셨습니다 ~~
마지막으로 말씀드리자면 포스터가 없는 대학은 아직도 포스터 영역이 사라지는데 의도하신 걸까요? 
이 부분만 명확하게 한다면 좋을 것 같습니다 !

etama123 commented:
> 긴 시간 동안 고생 많으셨습니다 ~~ 마지막으로 말씀드리자면 포스터가 없는 대학은 아직도 포스터 영역이 사라지는데 의도하신 걸까요? 이 부분만 명확하게 한다면 좋을 것 같습니다 !

맞습니다~ 
포스터가 없는데 해당 영역에 다른 이미지를 띄우는 것이 사용자에게 아직 로딩되지 않은건가? 하는 혼란을 줄 수도 있지 않을까 하고 생각했어요. 혹시 좋은 방법이 있으시면 나중에라도 말씀해주세요!!

etama123 committed:
Merge branch 'develop' into feat/11

etama123 committed:
style(MainActivity, HomeFragment, etc): 코드 포맷팅 및 스타일 가이드 적용

etama123 committed:
style(HomeArtistItem, HomeLineupItem, HomePosterList, FestabookImage): 코드 포맷팅 및 스타일 개선

프로젝트 전반의 UI 컴포저블 코드를 정리하고, 불필요한 임포트 제거 및 일관된 코드 스타일을 적용했습니다.

- **`FestabookImage.kt`:**
    - `modifier`와 `imageUrl` 파라미터의 순서를 변경하고 코드 가독성을 위해 들여쓰기를 조정했습니다.
    - `CoilImage` 내부의 `modifier` 설정을 `Modifier.fillMaxSize()`로 단순화했습니다.
    - `FestabookImageZoomPopup` 및 미리보기(Preview) 함수의 접근 제어자를 `private`으로 수정했습니다.
    - 불필요한 빈 줄을 제거하고 다중 라인 호출 구조를 정돈했습니다.

- **`HomePosterList.kt`:**
    - 더 이상 사용되지 않는 `CoilImage` 임포트를 제거했습니다.
    - `Modifier` 체이닝의 줄바꿈과 쉼표 누락 등 스타일을 수정했습니다.

- **`HomeLineupItem.kt` & `HomeArtistItem.kt`:**
    - 사용하지 않는 `DividerDefaults` 임포트를 제거했습니다.
    - `RoundedCornerShape` 정의 및 컴포저블 내 불필요한 공백을 제거하여 코드 포맷을 정돈했습니다.

etama123 committed:
refactor(Home): 디자인 시스템 적용 및 Fragment 주입 방식 리팩토링

etama123 committed:
test(HomeViewModelTest): LiveData를 StateFlow로 전환함에 따른 테스트 코드 리팩토링
